modules/p2/ding_list/include/form.inc

```diff
@@ -568,6 +569,24 @@ The link is active for 7 days from today, after which it will be disabled for se
     }
   }
 
+  // Information when public.
+  $form['public'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Public'),
+    '#description' => t('You can find your public list under !link', array('!link' => l(t('Public lists'), 'public-lists'))),
+    '#states' => array(
+      'visible' => array(
+        ':input[name="status"]' => array('value' => 'public'),
+      ),
+    ),
+    '#tree' => TRUE,
+  );
+  $form['public']['link'] = array(
+    '#type' => 'textfield',
```

Comment:
Why is this a textfield? The user is not able to change the value of it. If it is to be a textfield then I would expect it to be disabled. Otherwise use [a `item` type](https://api.drupal.org/api/drupal/developer%21topics%21forms_api_reference.html/7.x#item) or the like.